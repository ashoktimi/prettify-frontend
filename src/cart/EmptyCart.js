import React, { useState, useEffect } from "react";
import { addToCart } from "./cartSlice";
import PrettifyApi from "../api/api";
import LoadingSpinner from "../helpers/LoadingSpinner";
import { useDispatch } from "react-redux";
import { Stack } from "react-bootstrap";

const EmptyCart = () => {
    const [products, setProducts] = useState(null);
    const dispatch = useDispatch();
    useEffect(function getProductsOnMount() {
        search();
      }, []);

    async function search() {
        let products = await PrettifyApi.getRecProducts();
        setProducts(products.products);
    }
  const handleAddToCart = (product) => {
    dispatch(addToCart(product));
  };
  
  if (!products) return <LoadingSpinner />;
  return (
  <>
    <p>Your cart is empty</p>
      <p>Why not check out these recommended products?</p>
        <Stack gap={3}>
        {products.map(item => (
          <Stack gap={2}>
            <Stack direction="horizontal">
            <img
              src={item.image_link}    
              style={{ width: "125px", height: "115px", objectFit: "cover" }}
              alt={''}
            />
              {item.name}{" "}${(item.price)}
            </Stack> 
            
              <Stack direction="vertical" >         
  
                <div style={{display: "flex", justifyContent: 'center'}}>
                  <button onClick={() => handleAddToCart(item)} style={{border:'none', backgroundColor:'white',  marginRight: '15px'}}> add to cart </button>
                 
              </div>   
              </Stack>
          </Stack>
  
        ))}
      </Stack>
      </>
  );
};

export default EmptyCart;
