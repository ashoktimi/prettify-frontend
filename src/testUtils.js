import React from "react";
import UserContext from "./auth/userContext";


const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, handleToggleOffcanvas = () => false }) => (
    <UserContext.Provider value={{ currentUser, handleToggleOffcanvas }}>
      {children}
    </UserContext.Provider>
);

function setup(){
  window.matchMedia = jest.fn().mockImplementation(query => {
    return {
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  });
}

function teardown(){
   window.matchMedia.mockClear();
   delete window.matchMedia;
}

function commonBeforeEach() {
  setup();
}

function commonAfterEach() {
  teardown();
}

export { UserProvider, commonBeforeEach, commonAfterEach };