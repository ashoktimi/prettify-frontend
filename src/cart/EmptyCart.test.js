import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { commonBeforeEach, commonAfterEach } from "../testUtils";
import { configureStore } from '@reduxjs/toolkit';
import cartReducer from './cartSlice';
import EmptyCart from "./EmptyCart";


beforeEach(commonBeforeEach)
afterEach(commonAfterEach)

describe('EmptyCart', () => {
 const store = configureStore({
   reducer: {
     cart: cartReducer
   }
 });

 test("renders without crashing", function () {
    render(
      <MemoryRouter>
        <Provider store={store}>
            <EmptyCart />
        </Provider>    
      </MemoryRouter>   
     );
  });
  
  test("matches snapshot", function () {
    const { asFragment } = 
     render(
      <MemoryRouter>
        <Provider store={store}>
           <EmptyCart />       
        </Provider>    
      </MemoryRouter>  
     );
    expect(asFragment()).toMatchSnapshot();
  });

})