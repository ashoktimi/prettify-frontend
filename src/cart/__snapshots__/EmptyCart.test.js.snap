// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCart matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoadingSpinner"
  >
    Loading ...
  </div>
</DocumentFragment>
`;

exports[`ProductCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoadingSpinner"
  >
    Loading ...
  </div>
</DocumentFragment>
`;
